src/odemis/driver/smaract.py

```diff
@@ -4913,6 +4922,406 @@ def _cancelReference(self, future):
         return True
 
 
+class PicoscalePythonSDK(model.HwComponent):
+    """
+    A driver for a SmarAct Picoscale interferometer system using the Python SDK.
+
+    The device does not contain any actuators. Its main functionality is to provide the position for each
+    of its channels through a VA.
+
+    Attributes
+    ==========
+    .position (VA: str --> float): position in m for each channel
+    .referenced (VA: str --> bool): indicates which channels have been referenced
+
+    Functions
+    =========
+    .reference: performs adjustment routine required for precise position values
+    """
+
+    def __init__(self, name, role, locator, channels, ref_on_init=False, precision_mode=0, *args, **kwargs):
+        """
+        name: (str)
+        role: (str)
+        locator: (str)
+            For a real device, Picoscale controllers with USB interface can be addressed with the
+            following locator syntax:
+                usb:ix:<id>
+            where <id> is the first part of a USB devices serial number which
+            is printed on the Picoscale controller.
+            If the controller has a TCP/IP connection, use:
+                network:<ip>:<port>
+            The device can also be addressed by its serial number:
+                usb:sn:<serial_number>
+                network:sn:<serial_number>
+        channels: (str --> int) dictionary mapping channel names to channel numbers
+        ref_on_init: (True, False, "always", "if necessary", "never")
+            * "always": Run referencing procedure every time the driver is initialized, no matter the state it was in.
+            * True / "if necessary": If the channels are already in a valid state (i.e. the device was not turned off
+                since the last referencing), don't reference again. Only reference if the channels are not valid (i.e.
+                the device was turned off after the last referencing). In any case, the device can be used after
+                the referencing procedure is complete. It is recommended to reference the system frequently though.
+                In case the system has not been power cycled in a long time, the referencing parameters might become
+                outdated and the reported position values might not be accurate.
+            * False / "never": Never reference. This means that the device might not be able to produce position data,
+                if it was not previously referenced.
+        precision_mode: (0 <= int <= 5) strength of digital lowpass filter, a higher level corresponds to higher
+            precision, but lower velocity. Not available on all systems.
+        """
+        model.HwComponent.__init__(self, name, role, *args, **kwargs)
+
+        if not smaract_python_sdk:
+            raise ModuleNotFoundError("Smaract Python SDK modules are missing. Cannot run PicoScale driver.")
```

Comment:
When running the backend without the python SDK installed, does it bring up a good error message? (I'm mostly worried that the `ModuleNotFoundError` convinces the backend to show a generic error message, preventing to see the actual error. In this case you could raise just a OSError). 
Also, please add to the message the line to type to install the packages. `sudo apt install python3-smaract-picoscale...`.